rclcpp/include/rclcpp/node.hpp

```diff
@@ -284,7 +285,7 @@ class Node : public std::enable_shared_from_this<Node>
     const std::string & topic_name,
     CallbackT && callback,
     size_t qos_history_depth,
-    rclcpp::callback_group::CallbackGroup::SharedPtr group = nullptr,
+    rclcpp::callback_group::CallbackGroup::SharedPtr group,
```

Comment:
This breaks API though right? You cannot do this to resolve the ambiguity.

Reply:
I'm not sure I follow what you mean by "break the api" here. I have redirected existing calls to `topic_name, callback, qos_history_depth` through the new API - so they're compiling against a new function. But in `node_impl.hpp` I have directed all three interfaces through the new Options-based interface, so they do all receive the same functionality.